src/devices/Lps25h/Lps25h.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Diagnostics;
+using System.Device.I2c.Drivers;
+using System.Buffers.Binary;
+
+namespace Iot.Device.Lps25h
+{
+    public class Lps25h : IDisposable
+    {
+        private const byte ReadMask = 0x80;
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// Lps25h - Pressure and temperature sensor
+        /// </summary>
+        public Lps25h(I2cDevice i2cDevice)
+        {
+            _i2c = i2cDevice;
+
+            // Highest resolution for both pressure and temperature sensor
+            byte resolution = Read(Register.ResolutionMode);
+            resolution |= 0b1111;
+            WriteByte(Register.ResolutionMode, resolution);
+
+            byte control1orig = Read(Register.Control1);
+            // 7 - PD - power down control - 1 means active
+            // 6-4 - output data rate - 0b100 means 25Hz for both sensors
+            // 3 - interrupt circuit enable - 0 means disabled
+            // 2 - block data update - 1 means update when both MSB and LSB are read
+            // 1 - reset auto-zero - 0 means disable
+            // 0 - SPI mode - we don't care what value since we use I2c, leave at default (0)
+            byte control1 = 0b11000100;
+            WriteByte(Register.Control1, control1);
+        }
+
+        /// <summary>
+        /// Temperature in Celsius
+        /// </summary>
+        public float Temperature => 42.5f + ReadInt16(Register.Temperature) / 480f;
+
+        /// <summary>
+        /// Pressure in hPa
+        /// </summary>
+        public float Pressure => ReadInt24(Register.Pressure) / 4096f;
+
+        private void WriteByte(Register register, byte data)
+        {
+            Span<byte> buff = stackalloc byte[2]
+            {
+                (byte)register,
+                data
+            };
+
+            _i2c.Write(buff);
+        }
+
+        private static int ReadInt24LittleEndian(ReadOnlySpan<byte> buff)
+        {
+            Debug.Assert(buff.Length == 3);
+
+            Span<byte> b = stackalloc byte[4]
+            {
+                buff[0],
+                buff[1],
+                buff[2],
```

Comment:
Cache `buff[2]` in a local, up front?
It will probably remove the bounds checks on `buff[0]`, `buff[1]` later on as well.